bastion/src/macros.rs

```diff
@@ -0,0 +1,206 @@
+                .with_callbacks($cbs)
+                .with_redundancy($red)
+                .with_exec(|ctx: $crate::context::BastionContext| {
+                    async move {
+                        let ctx = ctx;
+                        loop {
+                            let msg = ctx.recv().await?;
+                            ($action)(msg);
+                        }
+                    }
+                })
+        }).expect("failed to create children group");
+    };
+    (@sort, $red:expr, $cbs:expr, $action:expr, $sp:expr,) => {
+        $sp.children(|ch| {
+            ch
+                .with_callbacks($cbs)
+                .with_redundancy($red)
+                .with_exec(|ctx: $crate::context::BastionContext| {
+                    async move {
+                        let ctx = ctx;
+                        loop {
+                            let msg = ctx.recv().await?;
+                            ($action)(msg);
+                        }
+                    }
+                })
+        }).expect("failed to create children group");
+    };
+}
+
+/// This macro creates a new supervisor with the given strategy and the given callbacks.
+/// Children can be specified by using the `children` / `child` macro.
+/// You can provide as many children groups as you want. Supervised supervisors are currently not
+/// supported.
+///
+/// # Example
+/// ```
+/// # use bastion::prelude::*;
+/// # fn main() {
+/// let sp = supervisor! {
+///     callbacks: Callbacks::default(),
+///     strategy: SupervisionStrategy::OneForAll,
+/// };
+/// # }
+/// ```
+#[macro_export]
+macro_rules! supervisor {
+    ($($keys:ident: $vals:expr,)*) => {
+        supervisor!(@sort,
+            $crate::supervisor::SupervisionStrategy::OneForAll,
+            $crate::Callbacks::default(),
+            $($keys: $vals,)*
+        )
+    };
+
+    (@sort,
+    $strat:expr, $_:expr,
+    callbacks: $cbs:expr,
+    $($keys:ident: $vals:expr,)*) => {
+        supervisor!(@sort,
+            $strat,
+            $cbs,
+            $($keys: $vals,)*
+        )
+    };
+
+    (@sort,
+    $_:expr, $cbs:expr,
+    strategy: $strat:expr,
+    $($keys:ident: $vals:expr,)*) => {
+        supervisor!(@sort,
+            $strat,
+            $cbs,
+            $($keys: $vals,)*
+        )
+    };
+
+    (@sort, $strat:expr, $cbs:expr,) => {
+        $crate::Bastion::supervisor(|sp| {
+            sp
+                .with_callbacks($cbs)
+                .with_strategy($strat)
+        }).expect("failed to create supervisor");
+    };
+}
+
+/// Spawnes a blocking task, which will be run on an extra thread pool, and
+/// returns the handle.
+///
+/// # Example
+/// ```
+/// # use std::{thread, time};
+/// # use lightproc::proc_stack::ProcStack;
+/// # use bastion::prelude::*;
+/// # fn main() {
+/// let task = blocking! {
+///     thread::sleep(time::Duration::from_millis(3000));
+/// };
+/// bastion_executor::run::run(task, ProcStack::default());
```

Comment:
So the `run` method is basically `block_on` for any other systems around.
If possible let's create `run` macro too, I forgot that.